de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/editor/annotations/AnnotationManager.java

```diff
@@ -35,14 +35,18 @@
         SELECTION_ANNOTATION, CONTRIBUTION_ANNOTATION
     }
 
+    private static final int MAX_CONTRIBUTION_ANNOTATIONS = Integer
+        .getInteger("saros.intellij.MAX_CONTRIBUTION_ANNOTATIONS",50);
+
     private final AnnotationStore<SelectionAnnotation> selectionAnnotationStore;
-    private final AnnotationStore<ContributionAnnotation> contributionAnnotationStore;
+    private final QueueAnnotationStore<ContributionAnnotation> contributionAnnotationStore;
```

Comment:
The variable name should imply that a queue/buffer is used, otherwise it is hard to understand in line 167 that the removeIfFull method returns a "dequeueAnnotation".